src/TerminalReader.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace PhpSchool\Terminal;
+
+/**
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class TerminalReader
```

Comment:
So this new class takes a `Terminal` object and allows to read input and maps to `InputCharacters` - it also allows to map other characters to control code. So `cli-menu` code will look like this:

```php
$reader = new TerminalReader($this->terminal);
$reader->addControlMappings([
    '^P' => InputCharacter::UP,
    'k'  => InputCharacter::UP,
    '^K' => InputCharacter::DOWN,
    'j'  => InputCharacter::DOWN,
    "\r" => InputCharacter::ENTER,
    ' '  => InputCharacter::ENTER,
]);

while ($this->isOpen() && $char = $reader->readCharacter()) {
    if ($char->isNotControl()) {
        continue;
    }

    switch ($char->getControl()) {
        case InputCharacter::UP:
        case InputCharacter::DOWN:
            $this->moveSelection($char->getControl());
            $this->draw();
            break;
        case InputCharacter::ENTER:
            $this->executeCurrentItem();
            break;
    }
}
```
What do you think?